src/Pos/Wallet/Web/Server/Methods.hs

```diff
@@ -451,22 +451,25 @@ sendExtended sendActions cpassphrase srcCAddr dstCAddr c curr title desc = do
 getHistory
     :: forall ssc m.
        (SscHelpersClass ssc, WalletWebMode ssc m)
-    => CAccountAddress -> Maybe Word -> Maybe Word -> m ([CTx], Word)
-getHistory cAccAddr skip limit = do
-    (minit, cachedTxs) <- transCache <$> getHistoryCache cAccAddr
-    cAddr <- decodeCAddressOrFail $ caaAddress cAccAddr
+    => CWalletAddress -> Maybe Word -> Maybe Word -> m ([CTx], Word)
+getHistory wAddr skip limit = do
+    cAccAddrs <- getWalletAccAddrsOrThrow wAddr
+    cHistory <- fmap concat . forM cAccAddrs $ \cAccAddr -> do
```

Comment:
Seems like [`concatForM`](https://github.com/serokell/universum/blob/f2ccf8afd862e37ccd204c0ef9efde48a05c2d29/src/Monad.hs#L93) can be used here